File: src/main/java/net/aspw/nightx/features/module/modules/movement/VanillaFlight.java
package net.aspw.nightx.features.module.modules.movement;

import net.aspw.nightx.event.EventTarget;
import net.aspw.nightx.event.UpdateEvent;
import net.aspw.nightx.features.module.Module;
import net.aspw.nightx.features.module.ModuleCategory;
import net.aspw.nightx.features.module.ModuleInfo;

@ModuleInfo(name = "VanillaFlight", spacedName = "Vanilla Flight", category = ModuleCategory.MOVEMENT)
public class VanillaFlight extends Module {

    @EventTarget
    public void onUpdate(final UpdateEvent event) {
        mc.thePlayer.capabilities.allowFlying = true;
    }

    @Override
    public void onDisable() {
        mc.thePlayer.capabilities.allowFlying = false;
        mc.thePlayer.capabilities.isFlying = false;

        if (mc.thePlayer.capabilities.isCreativeMode)
            mc.thePlayer.capabilities.allowFlying = true;
    }
}

File: src/main/java/net/aspw/nightx/features/module/modules/movement/VanillaFlight.kt
package net.aspw.nightx.features.module.modules.movement

import net.aspw.nightx.event.EventTarget
import net.aspw.nightx.event.UpdateEvent
import net.aspw.nightx.features.module.Module
import net.aspw.nightx.features.module.ModuleCategory
import net.aspw.nightx.features.module.ModuleInfo

@ModuleInfo(name = "VanillaFlight", spacedName = "Vanilla Flight", category = ModuleCategory.MOVEMENT)
class VanillaFlight : Module() {
    @EventTarget
    fun onUpdate(event: UpdateEvent?) {
        mc.thePlayer.capabilities.allowFlying = true
    }

    override fun onDisable() {
        mc.thePlayer.capabilities.allowFlying = false
        mc.thePlayer.capabilities.isFlying = false
        if (mc.thePlayer.capabilities.isCreativeMode) mc.thePlayer.capabilities.allowFlying = true
    }
}